scripts/profile.js

```diff
@@ -53,11 +53,156 @@ function listGoals() {
 	);
 }
 
-// First, log in…
-firebase.auth().onAuthStateChanged(function (user) {
-        if (user) { 
-			// Now we have a reference to the logged in user!
-			// Update the page, list the goals of the currently signed-in user.
+// The variable which stores the UID to be used as a reference in multiple methods
+var currentUser;
+
+//Fill the profile with the data in fireDB.
+function populateInfo() {
+	firebase.auth().onAuthStateChanged(user => {
+		// Check if user is signed in:
+		if (user) {
+			//Assign UID to currentUser.
+			currentUser = db.collection("users").doc(user.uid);
+
+			//go to the correct user document by referencing to the user uid
+			//get the document for current user.
+			currentUser.onSnapshot(userDoc => {
+				//get the data fields of the user
+				var userName = userDoc.data().name;
```

Comment:
Try to use `let` instead of `var`, helps avoid weird one-off scoping issues.

Reply:
rog